### PR TITLE
[WIP - FEEDBACK] initial try for tinyMCE wysiwyg field in new interface

### PR DESCRIPTION
**Pull request by gautamsi**
Trying to add the Wysiwyg field in new interface. It seems to be working with a caveat that I could not figure out the override config to pass.

@timleslie @JedWatson  can you provide some feedback on this please. I am adding comment on what I could not figure out.

hopefully we can land this soon.

### CODE REVIEW

File: packages-next/keystone/src/lib/createExpressServer.ts [hunk 1]
```diff
@@ -82,6 +82,17 @@ export const createExpressServer = async (
   const { graphQLSchema, createContext } = system;
   addApolloServer({ server, graphQLSchema, createContext, sessionImplementation });
 
+  const apps = ([...system.keystone.registeredTypes])
```
Comment (gautamsi) [inline packages-next/keystone/src/lib/createExpressServer.ts:85]: I had to copy this from keystone prepare code which does not seems to be running anymore in new interface. Have to figure out how to optimize this. I am sure this is not the best way

File: packages-next/wysiwyg/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import path from 'path';
+// @ts-ignore
+import { Wysiwyg } from '@keystonejs/fields-wysiwyg-tinymce';
+import type { FieldType, FieldConfig, BaseGeneratedListTypes } from '@keystone-next/types';
+
+type WysiwygFieldConfig<
+  TGeneratedListTypes extends BaseGeneratedListTypes
+> = FieldConfig<TGeneratedListTypes> & {
+  isRequired?: boolean;
+  editorConfig: any;
```
Comment (gautamsi) [inline packages-next/wysiwyg/src/index.ts:10]: I am unable to figure out how to get this value working.

File: packages-next/wysiwyg/src/views/Field.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  onChange,
+}: FieldProps<typeof import('.').controller>) {
+
+  const handleChange = (value: string) => {
+    if (typeof value === 'string') {
+      onChange(value);
+    }
+  };
+
+  const overrideOptions = field.editorConfig;
```
Comment (gautamsi) [inline packages-next/wysiwyg/src/views/Field.tsx:46]: this value is never populated.

File: packages-next/wysiwyg/src/views/Field.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+import dynamic from 'next/dynamic';
+
+import { jsx, Stack, useTheme } from '@keystone-ui/core';
+
+import { FieldContainer, FieldLabel } from '@keystone-ui/fields';
+import { FieldProps } from '@keystone-next/types';
+
+// disable ssr for TinyMCE assets. also the 
+const Editor = dynamic(async () => {
```
Comment (gautamsi) [inline packages-next/wysiwyg/src/views/Field.tsx:11]: dynamic import with disabling SSR needed to render TinyMCE Editor

File: packages-next/wysiwyg/src/views/Field.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  value,
+  onChange,
+}: FieldProps<typeof import('.').controller>) {
+
+  const handleChange = (value: string) => {
+    if (typeof value === 'string') {
+      onChange(value);
+    }
+  };
+
```
Comment (gautamsi) [inline packages-next/wysiwyg/src/views/Field.tsx:45]: missing access control check which would not render this field but seems that is overall issue with new interface it seems.

File: packages-next/wysiwyg/src/views/index.tsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+export { Field } from './Field';
+
+export const Cell: CellComponent = ({ item, field }) => {
+  const data = item[field.path];
+  if (!data) return null;
+  return (data);
+};
+
+export const CardValue: CardValueComponent = ({ item, field }) => {
```
Comment (gautamsi) [inline packages-next/wysiwyg/src/views/index.tsx:20]: have not tested this yet. hopefully we can have a view only field and then an edit button to render tiny editor

**Comment by gautamsi** [comment]
@timleslie any feedback on this, should I complete this? specific issue you see?

**Comment by timleslie** [comment]
> @timleslie any feedback on this, should I complete this? specific issue you see?

I don't think we plan to include support for tinyMCE as part of the new interfaces. We're focussing our effort on the new `Document` field, which will provide support for advanced editor functionality.

**Comment by gautamsi** [comment]
thanks for the update, I will try add this to my contrib package then.

I will ping for help if having issue in creating separate package for this in TS there.

**Comment by gautamsi** [comment]
reopen if needed.

**Comment by gautamsi** [comment]
if anyone needs, it is moved here - https://github.com/keystonejs-contrib/keystonejs-contrib/pull/17

**Pull request closed without merging**